lib/autoproj/system.rb

```diff
@@ -6,6 +6,14 @@ class UserError < RuntimeError; end
     def self.create_symlink(from, to)
         if Autobuild.windows?
             Dir.create_junction(to, from)
+        elsif Autobuild.msys?
+            begin
+                #msys created a copy instead of a link
+                FileUtils.ln_sf from, to
+            rescue Errno::EEXIST
+                FileUtils.rm_rf to
+                FileUtils.ln_sf from, to
+            end
```

Comment:
This function is available in the win32-dir gem, but sadly, that is not compatible with msys2.